react/components/EXPERIMENTAL_Table/DataTable/Rows.tsx

```diff
@@ -18,62 +17,76 @@ const Rows: FC<RowsProps> = ({
-    const toggleChecked = () => checkboxes.toggle(rowData)
+  return items ? (
+    <>
+      {items.map((rowData, i) => {
+        const toggleChecked = () => checkboxes.toggle(rowData)
 
-    const isRowChecked = checkboxes && checkboxes.isChecked(rowData)
-    const isRowPartiallyChecked =
-      checkboxes && checkboxes.isPartiallyChecked(rowData)
-    const isRowSelected = isRowChecked || isRowPartiallyChecked
+        const isRowChecked = checkboxes && checkboxes.isChecked(rowData)
+        const isRowPartiallyChecked =
+          checkboxes && checkboxes.isPartiallyChecked(rowData)
+        const isRowSelected = isRowChecked || isRowPartiallyChecked
 
-    const clickable = onRowClick
-      ? {
-          onClick: () => onRowClick({ rowData }),
-        }
-      : {}
-    return (
-      <Row
-        {...rowProps}
-        {...clickable}
-        height={rowHeight}
-        active={(isRowActive && isRowActive(rowData)) || isRowSelected}
-        key={`${NAMESPACES.ROW}-${uuid()}`}>
-        {columns.map((column: Column, cellIndex: number) => {
-          const { cellRenderer, width } = column
-          const cellData = rowData[column.id]
-          const content = cellRenderer
-            ? cellRenderer({ cellData, rowData, rowHeight, selectedDensity })
-            : cellData
-          return (
-            <Row.Cell {...cellProps} key={`cel-${uuid()}`} width={width}>
-              {cellIndex === 0 && checkboxes && (
-                <CellPrefix>
-                  <span className="ph3">
-                    <CellPrefix.Checkbox
-                      checked={isRowChecked}
-                      partial={isRowPartiallyChecked}
-                      disabled={checkboxes.isDisabled(rowData)}
-                      onClick={toggleChecked}
-                    />
-                  </span>
-                </CellPrefix>
-              )}
-              {content}
-            </Row.Cell>
-          )
-        })}
-      </Row>
-    )
-  }
-
-  return <>{items && items.map(renderRow)}</>
+        const clickable = onRowClick
+          ? {
+              onClick: () => onRowClick({ rowData }),
+            }
+          : {}
+        return (
+          <Row
+            {...rowProps}
+            {...clickable}
+            height={rowHeight}
+            active={(isRowActive && isRowActive(rowData)) || isRowSelected}
+            key={rowKey({ rowData })}>
+            {columns.map((column: Column, cellIndex: number) => {
+              const { cellRenderer, width } = column
+              const cellData = rowData[column.id]
+              const content = cellRenderer
+                ? cellRenderer({
+                    cellData,
+                    rowData,
+                    rowHeight,
+                    selectedDensity,
+                  })
+                : cellData
+              return (
+                <Row.Cell
+                  {...cellProps}
+                  key={cellKey({ cellData })}
```

Comment:
you can use the `column.id` here, and avoid adding a new prop to the table component

Reply:
nice suggestion

Reply:
Done